### PR TITLE
Add eslint-plugin-import

### PR DESCRIPTION
**Pull request by ergunsh**
Includes some static checks that can prevent some mistakes (https://github.com/zeplin/zeplin-webapp/pull/1341) and some styling rules.

For styling rules, I tried to reflect on what we have been currently using without linting. So, I didn't add some rules that we are not currently following but may also consider using, those are:
* Report namespace imports ([`no-namespace`])
* Ensure consistent use of file extension within the import path ([`extensions`])
* Enforce a convention in module import order ([`order`])
* Enforce a newline after import statements ([`newline-after-import`])
* Prefer a default export if module exports a single name ([`prefer-default-export`])
* Limit the maximum number of dependencies a module can have ([`max-dependencies`])
* Forbid unassigned imports ([`no-unassigned-import`])
* Forbid named default exports ([`no-named-default`])
* Forbid default exports ([`no-default-export`])
* Forbid named exports ([`no-named-export`])
* Forbid anonymous values as default exports ([`no-anonymous-default-export`])
* Prefer named exports to be grouped together in a single export declaration ([`group-exports`])
* Enforce a leading comment with the webpackChunkName for dynamic imports ([`dynamic-import-chunkname`])

Personally, I like to see ordered imports 😅

Also, there is this project specific thing that may be useful for webapp:
* Restrict which files can be imported in a given folder ([`no-restricted-paths`]): we can prevent client modules importing nodejs modules in mw

[`no-restricted-paths`]: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-restricted-paths.md
[`no-namespace`]: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-namespace.md
[`extensions`]: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/extensions.md
[`order`]: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/order.md
[`newline-after-import`]: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/newline-after-import.md
[`prefer-default-export`]: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/prefer-default-export.md
[`max-dependencies`]: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/max-dependencies.md
[`no-unassigned-import`]: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-unassigned-import.md
[`no-named-default`]: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-named-default.md
[`no-anonymous-default-export`]: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-anonymous-default-export.md
[`group-exports`]: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/group-exports.md
[`no-default-export`]: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-default-export.md
[`no-named-export`]: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-named-export.md
[`dynamic-import-chunkname`]: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/dynamic-import-chunkname.md

### CODE REVIEW

File: base.js [hunk 1]
```diff
@@ -285,6 +288,37 @@ module.exports = {
...
       }
-    ]
+    ],
+    // Checks
+
+    // Ensure imports point to files/modules that can be resolved
+    // https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-unresolved.md
+    "import/no-unresolved": ["error", { commonjs: true, caseSensitive: true }],
+    // Ensure named imports coupled with named exports
+    // https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/named.md#when-not-to-use-it
```
Comment (merih) [inline base.js]: Why fragment in the link?

File: base.js [hunk 2]
```diff
@@ -285,6 +288,37 @@ module.exports = {
...
+    // https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/export.md
+    "import/export": "error",
+    // Style
+
+    // Report any imports that come after non-import statements.
+    // https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/first.md
+    "import/first": "error",
+    // Ensure all exports appear after other statements
+    // https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/exports-last.md
+    "import/exports-last": "error",
```
Comment (merih) [inline base.js]: Does this allow things like:

```
export default function Color() {}

Color.prototype.something = function () {}
```

?

Comment (ergunsh) [reply base.js]: Sadly no, we need to fix some.

Comment (merih) [reply base.js]: I like the way it looks in the example above? Does anyone have any preference?

Comment (dirtybit) [reply base.js]: I am more inclined to follow `exports-last` convention that the contradicting one since it makes exported names easy to see.

Comment (ergunsh) [reply base.js]: ping @merih

**Pull request merged**